packages/flutter_tools/test/general.shard/build_system/targets/dart_plugin_registrant_test.dart

```diff
@@ -0,0 +1,226 @@
+        fileSystem: fileSystem,
+        logger: BufferLogger.test(),
+        processManager: FakeProcessManager.any(),
+        generateDartPluginRegistry: true);
+
+    final File config = environment.projectDir
+        .childDirectory('.dart_tool')
+        .childFile('package_config.json');
+    config.createSync(recursive: true);
+    config.writeAsStringSync(Platform.isWindows ? _kSamplePackageJsonWindows : _kSamplePackageJson);
+
+    final File pubspec = environment.projectDir.childFile('pubspec.yaml');
+    pubspec.createSync();
+    final File packages = environment.projectDir.childFile('.packages');
+    packages.createSync();
+
+    final File generatedMain = environment.projectDir
+        .childDirectory('.dart_tool')
+        .childDirectory('flutter_build')
+        .childFile('generated_main.dart');
+
+    final FlutterProject testProject = FlutterProject.fromDirectoryTest(environment.projectDir);
+    await DartPluginRegistrantTarget.test(testProject).build(environment);
+
+    expect(generatedMain.existsSync(), isFalse);
+  });
+
+  testUsingContext('regenerates generated_main.dart', () async {
+    final FileSystem fileSystem = _getFileSystem();
+    final Environment environment = Environment.test(
+        fileSystem.currentDirectory,
+        projectDir: fileSystem.directory('project')..createSync(),
+        artifacts: null,
+        fileSystem: fileSystem,
+        logger: BufferLogger.test(),
+        processManager: FakeProcessManager.any(),
+        generateDartPluginRegistry: true);
+    final File config = environment.projectDir
+        .childDirectory('.dart_tool')
+        .childFile('package_config.json');
+    config.createSync(recursive: true);
+    config.writeAsStringSync(Platform.isWindows ? _kSamplePackageJsonWindows : _kSamplePackageJson);
+
+    final File pubspec = environment.projectDir.childFile('pubspec.yaml');
+    pubspec.createSync();
+    pubspec.writeAsStringSync(_kSamplePubspecFile);
+    final File packages = environment.projectDir.childFile('.packages');
+    packages.createSync();
+    packages.writeAsStringSync(Platform.isWindows ? _kSamplePackagesFileWindows : _kSamplePackagesFile);
+
+    final File generatedMain = environment.projectDir
+        .childDirectory('.dart_tool')
+        .childDirectory('flutter_build')
+        .childFile('generated_main.dart');
+    generatedMain.createSync(recursive: true);
+
+    final File pluginPubspec = environment.fileSystem.currentDirectory.childDirectory('path_provider_linux').childFile('pubspec.yaml');
+    pluginPubspec.createSync(recursive: true);
+    pluginPubspec.writeAsStringSync(_kSamplePluginPubspec);
+    final FlutterProject testProject = FlutterProject.fromDirectoryTest(environment.projectDir);
+    await DartPluginRegistrantTarget.test(testProject).build(environment);
+
+    final String mainContent = generatedMain.readAsStringSync();
+    expect(mainContent, contains(_kLinuxRegistrant));
+  });
+}
+
+FileSystem _getFileSystem() {
```

Comment:
Please used a fixed filesystem style for the test

Reply:
Done